### PR TITLE
Update bootloaders with IDF release/v5.4 ones

### PR DESCRIPTION
**Pull request by JurajSadel**
closes https://github.com/esp-rs/espflash/issues/853

### CODE REVIEW

**Comment by JurajSadel** [comment]
Some help with testing will be appreciated here 😅

**Comment by bjoernQ** [comment]
There is one catch: newer bootloaders _need_ the app-descriptor which we support via the unreleased esp-bootloader-esp-idf - users will need to use that to have a app-descriptor

I guess updating to the latest bootloader which doesn't require the app-descriptor would be the best option for now

**Comment by MabezDev** [comment]
> There is one catch: newer bootloaders need the app-descriptor which we support via the unreleased esp-bootloader-esp-idf - users will need to use that to have a app-descriptor
>
> I guess updating to the latest bootloader which doesn't require the app-descriptor would be the best option for now

but all v5.x versions need the app image desc, and no 4.x boot loader is in "maintenance", though the maintenance is probably not an issue.

I think we should switch now, and start tackling this breakage. We'll have various tests/CI workflows etc that will need to be updated. From that we can find edge cases and troubleshooting steps ready for a migration guide entry in esp-hal later (doesn't strictly belong there, but gets the most eyeballs). If we wait, we only delay this process until the espflash release.

**Comment by bjoernQ** [comment]
> but all v5.x versions need the app image desc

I _think_ some older 5.x bootloaders don't strictly require it (at least w/o OTA support). Let me have a look

**Comment by bjoernQ** [comment]
In any case we should document which exact version the bootloaders are from (ideally including the exact commit) and what config options were used (or maybe we should also define them here - see https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/kconfig.html#bootloader-config)

**Comment by elipsitz** [comment]
This could also be a good opportunity to fix the issue with the esp32-s3 bootloader (#657)

**Comment by bjoernQ** [comment]
> "I (76) esp_image: MMU page size mismatch, configured: 0x8000, found: 0x10000"

This is not great given we take all the effort to generate new bootloaders

esp-hal (for now) uses 0x10000 fixed/hard-coded (applies to C2, C6, H2 IIRC)

File: espflash/resources/bootloaders/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+The listed bootloaders were built with [esp-idf/release/v5.4](https://github.com/espressif/esp-idf/tree/release/v5.4) at commit `3ad3632`, using the default settings.
```
Comment (bjoernQ) [inline espflash/resources/bootloaders/README.md:1]: Maybe pasting the output of `grep BOOTLOADER sdkconfig` would be even better

Comment (JurajSadel) [reply espflash/resources/bootloaders/README.md]: Good idea

**Comment by jessebraham** [comment]
> > "I (76) esp_image: MMU page size mismatch, configured: 0x8000, found: 0x10000"
> 
> This is not great given we take all the effort to generate new bootloaders
> 
> esp-hal (for now) uses 0x10000 fixed/hard-coded (applies to C2, C6, H2 IIRC)

Probably we should build the bootloaders with the same values being used by the HAL, in this case.

**Comment by JurajSadel** [comment]
> > > "I (76) esp_image: MMU page size mismatch, configured: 0x8000, found: 0x10000"
> > 
> > 
> > This is not great given we take all the effort to generate new bootloaders
> > esp-hal (for now) uses 0x10000 fixed/hard-coded (applies to C2, C6, H2 IIRC)
> 
> Probably we should build the bootloaders with the same values being used by the HAL, in this case.

I rebuilt C2, C6, and H2 with `0x10000` so the warning should be gone

**Comment by MabezDev** [comment]
> This is not correct. The published version of espflash is using the bootloader from v5.1 and does not require it. IIRC v5.2 also does not require it. The dependency on the application descriptor was added with either v5.3 or v5.4 (I don't recall which).

Thanks for clearing that up, I thought it was all 5.x versions :sweat_smile:. We could only update to v5.2 here, but I think we should just rip this band aid off as this app desc thing will be a requirement going forward if we want to update the bootloaders again in the future.

File: espflash/resources/bootloaders/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+The listed bootloaders were built with [esp-idf/release/v5.4](https://github.com/espressif/esp-idf/tree/release/v5.4) at commit `3ad3632`, using default settings:
```
Comment (jessebraham) [inline espflash/resources/bootloaders/README.md:1]: Are these default settings the same for all chips?

**Review by jessebraham — APPROVED** [review]
Thanks, LGTM!
